### DISCUSSION TITLE
Apply Material-UI ThemeProvider in Docs?

### DISCUSSION THREAD

**Discussion by nwpappas** [storybookjs/storybook#17937] [Help]
How can I apply a Material-UI theme, using the `ThemeProvider`, in the Doc's preview `iframe`?

I am currently applying the theme into the `Story` using a `decorator`...

```javascript
import { CssBaseline, ThemeProvider } from '@material-ui/core';
import { useDarkMode } from 'storybook-dark-mode';

// ... snip

export const decorators = [
  Story => (
    <ThemeProvider theme={useDarkMode() ? darkTheme : lightTheme}>
      <CssBaseline>
        <Story />
      </CssBaseline>
    </ThemeProvider>
  )
];
```

... but I'd like to apply it over the entire Docs pane so I can include our components outside a story and have them themed correctly.

Is there a decorator, or similar, for the entire Doc's `iframe`?

**Comment by Brian-McBride** [comment]
You can put it in the `preview.js` file and it will apply to all stories. 

```
import { addDecorator } from '@storybook/react';

// other code

addDecorator(
  (Story) => (
    <ThemeProvider theme={createTheme(useDarkMode() ? themeDark : themeLight)}>
      <CssBaseline />
      <Story />
    </ThemeProvider>
  ),
)
```
I don't think you put your story inside the CssBaseline tag?

**Reply by nwpappas** [reply]
Thanks @Brian-McBride.

I'm currently adding the decorator the the Story without trouble. The code snippet I included is what I am currently using and it is working okay.

What I'm wanting to do is apply the same styling _outside_ a Story, so I can include a component outside a story (as part of the documentation) and have it styled appropriately. I'm needing to target the `iframe` the docs reside in, the `html` that sites just inside, or the `body`.

Appreciate the tip!
